PWGLF/Tasks/cascadecorrelations.cxx

```diff
@@ -0,0 +1,234 @@
+using FullTracksExtIU = soa::Join<aod::TracksIU, aod::TracksExtra, aod::TracksCovIU, aod::TracksDCA>;
+using FullTracksExtWithPID = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksCov, aod::TracksDCA, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr>;
+using FullTracksExtIUWithPID = soa::Join<aod::TracksIU, aod::TracksExtra, aod::TracksCovIU, aod::TracksDCA, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr>;
+
+// Add a column to the cascdataext table: IsSelected.
+// 0 = not selected, 1 = Xi, 2 = Omega
+namespace o2::aod
+{
+namespace cascadeflags
+{
+DECLARE_SOA_COLUMN(IsSelected, isSelected, int); //~!
+} // namespace cascadeflags
+DECLARE_SOA_TABLE(CascadeFlags, "AOD", "CASCADEFLAGS", //!
+                  cascadeflags::IsSelected);
+using CascDataExtSelected = soa::Join<CascDataExt, CascadeFlags>;
+} // namespace o2::aod
+
+struct cascadeSelector {
+  Produces<aod::CascadeFlags> cascflags;
+
+  // histo's
+  HistogramRegistry registry{
+    "registry",
+    {
+      {"hMassXiMinus", "hMassXiMinus", {HistType::kTH1F, {{3000, 0.0f, 3.0f, "Inv. Mass (GeV/c^{2})"}}}},
+    },
+  };
+
+  void process(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, aod::CascDataExt const& Cascades, aod::V0sLinked const&, aod::V0Datas const&, FullTracksExtIUWithPID const&)
+  {
+    for (auto& casc : Cascades) {
+      auto v0 = casc.v0_as<o2::aod::V0sLinked>();
+      if (!(v0.has_v0Data())) {
+        cascflags(0);
+        continue; // reject if no v0data
+      }
+      auto v0data = v0.v0Data();
+
+      // Let's try to do some PID
+      // these are the tracks:
+      auto bachTrack = casc.bachelor_as<FullTracksExtIUWithPID>();
+      auto posTrack = v0data.posTrack_as<FullTracksExtIUWithPID>();
+      auto negTrack = v0data.negTrack_as<FullTracksExtIUWithPID>();
+
+      // PID
+      // Lambda check
+      if (casc.sign() < 0) {
+        // Proton check:
+        if (TMath::Abs(posTrack.tpcNSigmaPr()) > 3) {
+          cascflags(0);
+          continue;
+        }
+        // Pion check:
+        if (TMath::Abs(negTrack.tpcNSigmaPi()) > 3) {
+          cascflags(0);
+          continue;
+        }
+      } else {
+        // Proton check:
+        if (TMath::Abs(negTrack.tpcNSigmaPr()) > 3) {
+          cascflags(0);
+          continue;
+        }
+        // Pion check:
+        if (TMath::Abs(posTrack.tpcNSigmaPi()) > 3) {
+          cascflags(0);
+          continue;
+        }
+      }
+      // Bachelor check
+      if (TMath::Abs(bachTrack.tpcNSigmaPi()) < 3) {
+        if (TMath::Abs(bachTrack.tpcNSigmaKa()) < 3) {
+          // TODO: ambiguous! ignore for now
+          cascflags(0);
+          continue;
+        }
+        cascflags(1);
+        continue;
+      } else if (TMath::Abs(bachTrack.tpcNSigmaKa()) < 3) {
+        cascflags(2);
+        continue;
+      }
+      // if we reach here, the bachelor was neither pion nor kaon
+      cascflags(0);
+    } // cascade loop
```

Comment:
Do you plan to implement some topological selections in this part of the code?

Reply:
Yes, in the future I will implement that!